lld/COFF/Driver.cpp

```diff
@@ -2012,8 +2016,18 @@ void LinkerDriver::linkerMain(ArrayRef<const char *> argsArr) {
   config->ltoDebugPassManager = ltoDebugPM;
 
   // Handle /lldsavetemps
-  if (args.hasArg(OPT_lldsavetemps))
-    config->saveTemps = true;
+  if (args.hasArg(OPT_lldsavetemps)) {
+    for (const char *s : lldsaveTempsValues)
```

Comment:
Previously, we could pass an empty `SaveTempsArgs` to `addSaveTemps` to signify that we want the default - but now we need to duplicate the set of components here. Is this an issue? (I.e. I guess the tradeoff is whether we want another bool flag, or want to wrap `llvmsaveTemps` in `std::optional`, to signify whether it is wanted but the set is empty, or whether we explicitly want to just list all the default components here.)

Reply:
That's correct, `SaveTempsArgs` being empty on its own would be insufficient to disambiguate whether we mean the default or all components. We could move to `std::optional` as you suggest. I believe the main drawback is that the list of options has the potential to fall out of sync if it isn't explicit. The current approach forces developers keep the `/lldsavetemps:` options up-to-date should there be more temp files in the future.

I'll be happy to make the change whichever option is decided upon.

Reply:
Yeah, as long as we already have a list of the allowed elements (in order to error out early if passing an unsupported option) we can probably just go with it. (I guess the other alternative is to not error out early if passing unsupported values.

I'm ok with keeping this as is, I just wanted to talk through the various tradeoffs involved :-)

Also, I see that we already have the exact same analogue in the ELF driver, so that's another reason to keep things similar.

Reply:
Makes sense. Yes, I originally modeled it after the `ELF` driver trying to diverge only as necessary.